src/Compilers/CSharp/Portable/Binder/Binder_Symbols.cs

```diff
@@ -1161,7 +1161,7 @@ private Symbol UnwrapAlias(Symbol symbol, out AliasSymbol alias, BindingDiagnost
             {
                 var boundTypeArguments = BindTypeArguments(typeArguments, diagnostics, basesBeingResolved);
                 if (unconstructedType.IsGenericType
-                    && options.IsAttributeTypeLookup())
+                    && (options.IsAttributeTypeLookup() || containsAttributeType(node)))
```

Comment:
> && (options.IsAttributeTypeLookup() || containsAttributeType(node))) [](http://example.com/codeflow?start=19&length=69)

I think this is not the right place to perform this check. Not all constituent parts of the attribute type are going to be looked up at the attribute name location. For example,
```
using AB = A<(int A, int B)>.B;

[AB]
class C
{
}

class A<T>
{  
  public class B : System.Attribute
  {
  }
}
```

"AB" in ```[AB]``` is not even a generic name (so we will not get here), yet the attribute type still has the same problem. I think the check should be moved to where we bind attribute application. Once we get the attribute type back, we should simply run the validation across all of its parts, containing types, type arguments, etc. (VisitType does that, I think).

It looks like Binder.BindAttributeTypes method is the right place to do the check. Right after ```binder.BindType``` call.

Reply:
Sounds like a much better place, you are right. Moved the check there: cd02bbaf4f4ac4c2e8ff970cccea1897070bd9e4